pkg/storage/storage.go

```diff
@@ -142,6 +151,52 @@ func (s *Storage) History(name string) ([]*rspb.Release, error) {
 	return s.Driver.Query(map[string]string{"NAME": name, "OWNER": "TILLER"})
 }
 
+// removeLeastRecent removes items from history until the lengh number of releases
+// does not exceed max.
+//
+// We allow max to be set explicitly so that calling functions can "make space"
+// for the new records they are going to write.
+func (s *Storage) removeLeastRecent(name string, max int) error {
+	if max <= 0 {
```

Comment:
I think we want to keep at least one release rolling around. We're just limiting the maximum amount of releases down to a reasonable size. We don't want to completely remove the release ledger unless the user removed the release with `--purge`.